prinz_util/src/main/scala/pl/touk/nussknacker/prinz/util/http/AsyncRestJsonClient.scala

```diff
@@ -15,19 +16,18 @@ class AsyncRestJsonClient(baseUrl: String, private val backend: SttpBackend[Futu
   def postJsonBody[BODY, RESPONSE: Manifest](body: BODY, relativePath: String = "", params: RestRequestParams = EmptyRestRequestParams)
                                             (implicit encoder: Encoder[BODY], decoder: Decoder[RESPONSE]): Future[RestClientResponse[RESPONSE]] = {
     val request = createPostJsonRequest(relativePath, body, params)
-    logger.info("Sending request in {}: {}", getClass.getSimpleName, request)
-    wrapCaughtException(
-      () => request.send(backend).map { response => response.body.left.map(clientExceptionFromResponse) },
-      exception => Future(Left(RestClientException(exception.getMessage)))
-    )
+    wrap(request)
   }
 
   def getJson[RESPONSE: Manifest](relativePath: String = "", params: RestRequestParams = EmptyRestRequestParams)
                                  (implicit decoder: Decoder[RESPONSE]): Future[RestClientResponse[RESPONSE]] = {
     val request = createGetRequest(relativePath, params)
-    wrapCaughtException(
-      () => request.send(backend).map { response => response.body.left.map(clientExceptionFromResponse) },
-      e => Future(Left(RestClientException(e.getMessage)))
-    )
+    wrap(request)
   }
+
+  private def wrap[RESPONSE: Manifest]
+    (request: RequestT[Identity, Either[ResponseException[String, circe.Error], RESPONSE], Any]) = wrapCaughtException(
```

Comment:
also check if this method cannot be merged with the wrapCaughtException because now we have two wrapping exceptions methods which looks unnecessary